src/components/Badge/Badge.tsx

```diff
@@ -28,55 +28,90 @@ export enum BadgeStyle {
+              ]),
+              [BadgeStyle.Danger]: merge([
+                  "tw-bg-box-negative-strong tw-text-box-negative-strong-inverse",
+                  hasHover && "hover:tw-bg-box-negative-strong-hover hover:tw-text-box-negative-strong-inverse-hover",
+              ]),
+              [BadgeStyle.Progress]: merge([
+                  "tw-bg-box-selected-strong tw-text-box-selected-strong-inverse",
+                  hasHover && "hover:tw-bg-box-selected-strong-hover hover:tw-text-box-selected-strong-inverse-hover",
+              ]),
+              [BadgeStyle.Warning]: merge([
+                  "tw-bg-box-warning-strong tw-text-box-warning-strong-inverse",
+                  hasHover && "hover:tw-bg-box-warning-strong-hover hover:tw-text-box-warning-strong-inverse-hover",
+              ]),
+          }
+        : {
+              [BadgeStyle.Primary]: merge([
+                  "tw-bg-box-neutral tw-text-text-weak",
+                  hasHover && "hover:tw-bg-box-neutral-hover hover:tw-bg-box-neutral-inverse-hover",
+              ]),
+              [BadgeStyle.Positive]: merge([
+                  "tw-bg-box-positive tw-text-box-positive-inverse",
+                  hasHover && "hover:tw-bg-box-positive-hover hover:tw-text-box-positive-inverse-hover",
+              ]),
+              [BadgeStyle.Progress]: merge([
+                  "tw-bg-box-selected tw-text-bg-box-selected-inverse",
+                  hasHover && "hover:tw-bg-box-selected-hover hover:tw-text-box-selected-inverse-hover",
+              ]),
+              [BadgeStyle.Warning]: merge([
+                  "tw-bg-box-warning tw-text-box-warning-inverse",
+                  hasHover && "hover:tw-bg-box-warning-hover hover:tw-text-box-warning-inverse-hover",
+              ]),
+              [BadgeStyle.Danger]: merge([
+                  "tw-bg-box-negative tw-text-box-negative-inverse",
+                  hasHover && "hover:tw-bg-box-negative-hover hover:tw-text-box-negative-inverse-hover",
+              ]),
+          })[style] ?? "";
 
 const isBadgeStatus = (style: BadgeStatus | string): style is BadgeStatus =>
     Object.values(BadgeStatus).includes(style as BadgeStatus);
 
+const getSizeClasses = (children: ReactNode, hasExtra: boolean, isSmall: boolean) => {
+    const isCircular = !children && hasExtra;
+    const sizeClasses = [isSmall ? "tw-h-5" : "tw-h-6"];
+
+    if (isCircular) {
+        sizeClasses.push(isSmall ? "tw-w-5" : "tw-w-6");
+    } else {
+        sizeClasses.push(...[isSmall ? "tw-px-1.5" : "tw-px-2.5", isSmall ? "tw-gap-x-0.5" : "tw-gap-x-1"]);
+    }
+    return sizeClasses.join(" ");
+};
+
 export type BadgeProps = PropsWithChildren<{
     style?: BadgeStyle;
     icon?: ReactElement<IconProps>;
     status?: BadgeStatus | string;
     onClick?: () => void;
     onDismiss?: () => void;
+    disabled?: boolean;
+    emphasis?: "Strong" | "None";
+    size?: "s" | "m";
```

Comment:
Would it make sense to create Enums for `emphasis` and `size`?

Reply:
That is a good question. I will ask in the arcade team about this because I noticed some components deal exclusively in enums and others don't even touch them 🤔